spring-rabbit-stream/src/main/java/org/springframework/rabbit/stream/listener/StreamListenerContainer.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.rabbit.stream.listener;
+
+import org.springframework.amqp.core.Message;
+import org.springframework.amqp.core.MessageListener;
+import org.springframework.amqp.rabbit.listener.MessageListenerContainer;
+import org.springframework.amqp.rabbit.listener.api.ChannelAwareMessageListener;
+import org.springframework.beans.factory.BeanNameAware;
+import org.springframework.lang.Nullable;
+import org.springframework.rabbit.stream.support.converter.StreamMessageConverter;
+import org.springframework.util.Assert;
+
+import com.rabbitmq.stream.Consumer;
+import com.rabbitmq.stream.ConsumerBuilder;
+import com.rabbitmq.stream.Environment;
+import com.rabbitmq.stream.OffsetSpecification;
+
+/**
+ * A listener container for RabbitMQ Streams.
+ *
+ * @author Gary Russell
+ * @since 2.4
+ *
+ */
+public class StreamListenerContainer implements MessageListenerContainer, BeanNameAware {
+
+	private final Environment environment;
+
+	private final ConsumerBuilder builder;
+
+	private final StreamMessageConverter messageConverter = new StreamMessageConverter();
+
+	private java.util.function.Consumer<ConsumerBuilder> consumerCustomizer = c -> { };
+
+	private String stream;
+
+	private Consumer consumer;
+
+	private String listenerId;
+
+	private String beanName;
+
+	/**
+	 * Construct an instance using the provided environment.
+	 * @param environment the environment.
+	 */
+	public StreamListenerContainer(Environment environment) {
+		this.environment = environment;
+		this.builder = environment.consumerBuilder()
+				.offset(OffsetSpecification.first());
```

Comment:
Any reasons to use `OffsetSpecification#first` by default? The client default is `next` ([starting from the next offset to be written](https://rabbitmq.github.io/rabbitmq-stream-java-client/snapshot/htmlsingle/#specifying-an-offset)).

With `first` the consumer will start from the beginning of a stream. In case of application restart, a consumer that did not change the default will start again at the beginning of the stream.

Reply:
Thanks; will remove and set it in the test instead.